pkg/retriever/coordinators/race.go

```diff
@@ -20,18 +21,29 @@ func Race(ctx context.Context, queueOperations types.QueueRetrievalsFn) (*types.
 			waitGroup.Add(1)
 			go func() {
 				defer waitGroup.Done()
-				stats, err := nextRetrieval.Run()
+				stats, err := nextRetrieval.Run(ctx)
 				select {
 				case <-ctx.Done():
 				case resultChan <- types.RetrievalResult{Stats: stats, Err: err}:
 				}
 			}()
 		})
 	}()
+	done := make(chan struct{}, 1)
 	go func() {
 		waitGroup.Wait()
 		close(resultChan)
+		done <- struct{}{}
 	}()
+	stats, err := consume(ctx, resultChan)
+	select {
+	case <-done:
+	case <-time.After(100 * time.Millisecond):
```

Comment:
can we context cancel here? ... or even construct an `err` if one isn't set because, while this won't necessarily prevent a retrieval, it is an error as far as lassie is concerned

Reply:
context already cancelled but maybe we can track the error